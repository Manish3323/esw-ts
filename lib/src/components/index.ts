import CheckLogin from './aas/authentication/CheckLogin'
import ClientRole from './aas/authorization/ClientRole'
import RealmRole from './aas/authorization/RealmRole'
import { Consumer, AuthContext } from './aas/context/AuthContext'
import AuthContextProvider from './aas/context/AuthContextProvider'
import Login from './aas/Login'
import Logout from './aas/Logout'

// #export-components
export {
  Logout,
  Login,
  CheckLogin,
  RealmRole,
  ClientRole,
  AuthContextProvider,
  Consumer,
  AuthContext
}
// #export-components
