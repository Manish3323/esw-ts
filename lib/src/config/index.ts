export { AASConfig } from './AASConfig'
export { LocationConfig } from './LocationConfig'
export {
  AGENT_SERVICE_CONNECTION,
  AUTH_CONNECTION,
  CONFIG_CONNECTION,
  GATEWAY_CONNECTION,
  SEQUENCE_MANAGER_CONNECTION,
  resolveConnection,
  eswAgentConnection
} from './Connections'
export { setAppConfigPath } from './AppConfigPath'
