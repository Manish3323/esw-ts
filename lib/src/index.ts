export * from './clients/command'
export * from './clients/location'
export * from './clients/sequencer'
export * from './clients/alarm'
export * from './clients/aas'
export * from './clients/event'
export * from './clients/config'
export * from './models'
export * from './config'
export * from './components'
export * from './utils/download'
export type { TokenFactory } from './utils/TokenFactory'
