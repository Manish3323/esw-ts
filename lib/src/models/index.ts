export { ComponentId } from './ComponentId'
export type { ComponentType } from './ComponentType'
export { BaseKey } from './params/BaseKey'
export { ChoiceKeyFactory } from './params/ChoiceKeyFactory'
export { CurrentState } from './params/CurrentState'
export { Parameter } from './params/Parameter'
export { Prefix } from './params/Prefix'
export { Struct } from './params/Struct'
export type { Subsystem } from './params/Subsystem'
export type { Units } from './params/Units'
export { Result } from './params/Result'
export { Observe, SequenceCommand, ControlCommand, Setup, Wait } from './params/Command'
export type {
  OnewayResponse,
  SubmitResponse,
  Accepted,
  Cancelled,
  CancelledResponse,
  Completed,
  Error,
  ErrorResponse,
  InvalidResponse,
  CompletedResponse,
  Invalid,
  Locked,
  LockedResponse,
  Started,
  ValidateResponse,
  StartedResponse,
  CommandResponse
} from './params/CommandResponse'
export type {
  AltAzCoord,
  CometCoord,
  Coord,
  EqCoord,
  EqFrame,
  MinorPlanetCoord,
  RaDec,
  SolarSystemCoord,
  SolarSystemObject,
  ProperMotion,
  Tag
} from './params/Coord'

// export only types
export type {
  IntKey,
  LongKey,
  ShortKey,
  FloatKey,
  DoubleKey,
  ByteKey,
  StringKey,
  CharKey,
  StructKey,
  ChoiceKey,
  IntMatrixKey,
  ByteMatrixKey,
  LongMatrixKey,
  ShortMatrixKey,
  FloatMatrixKey,
  DoubleMatrixKey,
  IntArrayKey,
  ByteArrayKey,
  LongArrayKey,
  ShortArrayKey,
  FloatArrayKey,
  DoubleArrayKey,
  BooleanKey,
  UTCTimeKey,
  TAITimeKey,
  RaDecKey,
  EqCoordKey,
  SolarSystemCoordKey,
  MinorPlanetCoordKey,
  CometCoordKey,
  AltAzCoordKey,
  CoordKey,
  Key
} from './params/Key'

// helpers functions
export {
  intKey,
  longKey,
  shortKey,
  floatKey,
  doubleKey,
  byteKey,
  stringKey,
  charKey,
  booleanKey,
  byteMatrixKey,
  intMatrixKey,
  longMatrixKey,
  shortMatrixKey,
  floatMatrixKey,
  doubleMatrixKey,
  byteArrayKey,
  intArrayKey,
  longArrayKey,
  shortArrayKey,
  floatArrayKey,
  doubleArrayKey,
  structKey,
  utcTimeKey,
  taiTimeKey,
  choiceKey,
  raDecKey,
  eqCoordKey,
  solarSystemCoordKey,
  minorPlanetCoordKey,
  cometCoordKey,
  altAzCoordKey,
  coordKey
} from './params/Key'
