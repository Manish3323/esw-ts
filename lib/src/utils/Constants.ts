import type { AgentService } from '../clients/agent-service'
import type { AlarmService } from '../clients/alarm'
import type { CommandService } from '../clients/command'
import type { EventService } from '../clients/event'
import type { LocationService } from '../clients/location'
import type { SequenceManagerService } from '../clients/sequence-manager'
import type { AuthContextType } from '../components/aas/context/AuthContext'

export const APP_NAME = 'App-Name'

declare global {
  interface Window {
    isMocked?: boolean

    instance<T>(mockedValue: T): T

    mockedAlarmService: AlarmService
    mockedEventService: EventService
    mockedCommandService: CommandService
    mockedLocationService: LocationService
    mockedAgentService: AgentService
    mockedSequenceManager: SequenceManagerService
    auth: AuthContextType
  }
}
