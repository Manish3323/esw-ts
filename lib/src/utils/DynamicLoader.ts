export const dynamicImport = (path: string) => {
  return import(path)
}
