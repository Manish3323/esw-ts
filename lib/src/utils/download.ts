export const download = (object: any, fileName: string) => {
  const anchorElement = document.createElement('a')
  anchorElement.href = window.URL.createObjectURL(object)
  anchorElement.download = fileName
  document.body.appendChild(anchorElement)
  anchorElement.click()
  anchorElement.remove()
}
