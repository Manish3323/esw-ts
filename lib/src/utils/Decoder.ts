import { pipe } from 'fp-ts/lib/pipeable'
import * as D from 'io-ts/lib/Decoder'

export type Decoder<T> = D.Decoder<unknown, T>

export const char = pipe(
  D.string,
  D.refine((s): s is string => s.length == 1, 'single char')
)
