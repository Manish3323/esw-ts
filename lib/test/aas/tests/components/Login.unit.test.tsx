import React from 'react'
import { Provider } from '../../../../src/components/aas/context/AuthContext'
import { Login } from '../../../../src/components'
import { fireEvent, render } from '@testing-library/react'
import { mockAuth } from '../../../utils/MockHelpers'

// DEOPSCSW-630 - Javascript adapter for AAS
// DEOPSCSW-631 - React layer for javascript adapter for AAS
describe('<Login />', () => {
  test('should call login', async () => {
    const mockLogin = jest.fn()
    const { getByText } = await render(
      <Provider value={{ auth: mockAuth(), login: mockLogin, logout: jest.fn() }}>
        <Login />
      </Provider>
    )

    fireEvent.click(getByText('Login'))

    expect(mockLogin).toHaveBeenCalled()
  })

  test('should render login', async () => {
    const { container } = render(<Login />)
    expect(container).toMatchSnapshot()
  })
})
