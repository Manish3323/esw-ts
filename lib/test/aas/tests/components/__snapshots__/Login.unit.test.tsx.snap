// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render login 1`] = `
<div>
  <button
    id="aas-login"
  >
    Login
  </button>
</div>
`;
